axoupdater/src/lib.rs

```diff
@@ -453,34 +457,18 @@ impl AxoUpdater {
             installer_path
         };
 
-        // Before we update, move ourselves to a temporary directory.
+        // Before we update, rename ourselves to a temporary name.
         // This is necessary because Windows won't let an actively-running
         // executable be overwritten.
         // If the update fails, we'll move it back to where it was before
         // we began the update process.
-        // NOTE: this TempDir needs to be held alive for the whole function.
-        let temp_root;
         let to_restore = if cfg!(target_family = "windows") {
-            // If we know the install prefix, place the temporary directory for
-            // the executable there. This is because the `rename` syscall can't
-            // rename an executable across filesystem bounds, and there's a
-            // chance the system temporary directory could be on a different
-            // drive than the executable is.
-            // (A copy-and-delete move won't work because Windows won't let us
-            // delete a running executable - the same reason we're moving it out
-            // of the way to begin with!)
-            // See https://github.com/axodotdev/axoupdater/issues/120.
-            temp_root = if let Ok(prefix) = self.install_prefix_root() {
-                TempDir::new_in(prefix)?
-            } else {
-                TempDir::new()?
-            };
-            let old_path = std::env::current_exe()?;
-            let old_filename = old_path.file_name().expect("current binary has no name!?");
-            let ourselves = temp_root.path().join(old_filename);
-            std::fs::rename(&old_path, &ourselves)?;
+            let old_filename = std::env::current_exe()?;
 
-            Some((ourselves, old_path))
+            let mut new_filename = old_filename.as_os_str().to_os_string();
+            new_filename.push(OsStr::new(".__selfdelete__.exe"));
```

Comment:
is this name special..?

Reply:
Yes, it's mentioned here in the docs: https://docs.rs/self-replace/1.5.0/self_replace/#implementation I'll add a comment.